src/aa_remove_data/pb_utils.py

```diff
@@ -0,0 +1,164 @@
+from datetime import datetime, timedelta
+from os import PathLike
+
+from aa_remove_data.generated import EPICSEvent_pb2
+
+
+class PBUtils:
+    def __init__(self, filepath: PathLike | None = None):
+        """Initialise a PBUtils object. If filepath is set, read the protobuf
+        file at this location to gether its header, samples and type.
+
+        Args:
+            filepath (Optional[PathLike], optional): Path to pb file to be
+            read. Defaults to None.
+        """
+        self.header = EPICSEvent_pb2.PayloadInfo()  # type: ignore
+        self.samples = []
+        self.sample_type = ""
+        if filepath:
+            self.read_pb(filepath)
+
+    def _escape_data(self, data: bytes) -> bytes:
+        """Replace escape characters with alternatives, to avoid conflitcs.
+        Should exactly mirror _unescape_data().
+
+        Args:
+            data (bytes): A serialised protobuf sample.
+
+        Returns:
+            bytes: The serialised sample with escape characters replaced.
+        """
+        data = data.replace(b"\x1b", b"\x1b\x01")  # Escape escape character
+        data = data.replace(b"\x0a", b"\x1b\x02")  # Escape newline
+        data = data.replace(b"\x0d", b"\x1b\x03")  # Escape carriage return
+        return data
+
+    def _unescape_data(self, data: bytes) -> bytes:
+        """Replace escape character alternatives with the escape characters.
+        Should exactly mirror _escape_data().
+
+        Args:
+            data (bytes): A serialised protobuf message with escape characters
+            replaced.
+
+        Returns:
+            bytes: The serialised protobuf message containing escape
+            characters.
+        """
+        data = data.replace(b"\x1b\x03", b"\x0d")  # Unescape carriage return
+        data = data.replace(b"\x1b\x02", b"\x0a")  # Unescape newline
+        data = data.replace(b"\x1b\x01", b"\x1b")  # Unescape escape character
+        return data
+
+    def _convert_to_class_name(self, sample_type: str) -> str:
```

Comment:
Similarly I imagine `self.sample_type` can be used in place of the input. I imagine you have been using these functions directly for testing. We should try to convert this into a full API.

Reply:
As `self.sample_type` is initialised with blank chars, it may be worth adding some error checking to see if the variable has been assigned before working with the data.

Reply:
I've implemented this and changed the function name to `_get_proto_class_name()`. There's checking to see if `self.sample_type` (now called `self.pv_type`) has been assigned in `get_proto_class()` but maybe should be added here too?

Reply:
Looks good for now. Hopefully that will always be picked up in get_pv_type now. This will be ironed out when you add testing